janus_server/src/time.rs

```diff
@@ -0,0 +1,53 @@
+//! Utilities for timestamps and durations.
+
+use chrono::{naive::NaiveDateTime, Utc};
+use std::fmt::{Debug, Formatter};
+
+/// A clock knows what time it currently is.
+pub trait Clock: Clone + Debug + Sync + Send {
+    /// Get the current time.
+    fn now(&self) -> NaiveDateTime;
+}
+
+/// A real clock returns the current time relative to the Unix epoch.
+#[derive(Clone, Default)]
+pub struct RealClock {}
+
+impl Clock for RealClock {
+    fn now(&self) -> NaiveDateTime {
+        NaiveDateTime::from_timestamp(Utc::now().timestamp(), 0)
```

Comment:
Rounding to the second makes sense to me since that is the resolution supported by PPM `Time`, however the reason I used this form is because I didn't notice the one you suggested. This works so I'll leave it as is but I won't object to someone rewriting this module to make it simpler or more idiomatic.